opcua/common/xmlparser.py

```diff
@@ -341,8 +343,12 @@ def _parse_refs(self, el, obj):
             if ref.attrib["ReferenceType"] == "HasTypeDefinition":
                 obj.typedef = ref.text
             elif not struct.forward:
-                # Rmq: parent giving as ParentNodeId misses the ref type, so better to prioritize the ref descr
-                #if not obj.parent:
-                obj.parent = ref.text
-                if obj.parent == ref.text:
-                    obj.parentlink = ref.attrib["ReferenceType"]
+                parent, parentlink = struct.target, struct.reftype
+                if not obj.parent:
```

Comment:
in the next block I use the local variables to avoid the duplicated member and dictionary access..